libraries/AP_RAMTRON/AP_RAMTRON.h

```diff
@@ -12,25 +12,31 @@ class AP_RAMTRON {
     bool init(void);
 
     // get size in bytes
-    uint32_t get_size(void) const { return ramtron_ids[id].size_kbyte*1024UL; }
+    uint32_t get_size(void) const { return (id == UINT8_MAX) ? 0 : ramtron_ids[id].size_kbyte * 1024UL; }
 
     // read from device
-    bool read(uint32_t offset, uint8_t *buf, uint32_t size);
+    uint32_t read(uint32_t offset, uint8_t * const buf, uint32_t size);
```

Comment:
why change from bool when the only use case is to check if we get the full amount requested?

Reply:
It was useful while I was debugging to see why I wasn't getting back the data I expected. It also allows consumers to do partial retries.

Would you prefer this reverted back to returning a `bool`?

Reply:
yes, I'd prefer it be bool. If we end up with a consumer that wants to do partial retries in the future we could change this, but I doubt we will

Reply:
Reverted to `bool`.